src/main/java/io/quarkus/registry/app/DatabaseRegistryClient.java

```diff
@@ -173,14 +188,63 @@ private PlatformCatalog toPlatformCatalog(List<PlatformRelease> platformReleases
         return catalog.build();
     }
 
+    private List<ExtensionOrigin> toExtensionOrigins(ExtensionRelease extensionRelease) {
+        final List<ExtensionOrigin> extensionOrigins;
+
+        if (extensionRelease.platforms.isEmpty()) {
+            // Non-platform case
+            String id = getNonPlatformCoords(extensionRelease.quarkusCoreVersion);
+            extensionOrigins = List.of(ExtensionCatalog.builder().setId(id)
+                    .setQuarkusCoreVersion(extensionRelease.quarkusCoreVersion)
+                    .setPlatform(false)
+                    .build());
+        } else {
+
+            // Platform case
+            extensionOrigins = extensionRelease.platforms.stream()
+                    .map((platformExtension) -> (ExtensionOrigin) ExtensionCatalog.builder()
+                            // it should be <platform-key>:<member-bom-artifactId>-quarkus-platform-descriptor:<quarkus-version>:json:<quarkus-version>
+                            // We can get all member boms associated with this release, but not the particular one this extension came from
+                            // For the moment, we assume the member bom is quarkus-core since we don't have information to make a better choise
+                            .setId(ArtifactCoords
+                                    .of(platformExtension.platformRelease.platformStream.platform.platformKey,
+                                            QUARKUS_BOM + "-quarkus-platform-descriptor",
+                                            platformExtension.platformRelease.version,
+                                            Constants.JSON,
+                                            platformExtension.platformRelease.version)
+                                    .toGACTVString())
+                            .setBom(ArtifactCoords.pom(platformExtension.platformRelease.platformStream.platform.platformKey,
+                                    QUARKUS_BOM + "-quarkus-platform-descriptor",
```

Comment:
```suggestion
                                    QUARKUS_BOM,
```
Another alternative could be to construct the `quarkusBomCoords` and then call `PlatformArtifacts.getCatalogArtifactForBom(quarkusBomCoords)` to get the corresponding catalog coords.